actionpack/lib/action_controller/metal/redirecting.rb

```diff
@@ -143,6 +143,32 @@ def _compute_redirect_to_location(request, options) # :nodoc:
     module_function :_compute_redirect_to_location
     public :_compute_redirect_to_location
 
+    # Determine if it is safe to redirect to a location.
+    #
+    # This is useful if you have a redirection url come in from a request and,
+    # say, want to fallback to another URL if it isn’t safe to redirect there.
+    #
+    #   def return_url
+    #     if safe_to_redirect_to?(params[:return_to])
+    #       params[:return_to]
+    #     else
+    #       root_path
+    #     end
+    #   end
+    #
+    # A URL is considered safe if it is on the same host as the request URL.
+    # For example, if you’re serving a request to +example.com+:
+    #
+    #   safe_to_redirect_to?("https://example.com/profile")    # => true
+    #   safe_to_redirect_to?("http://example.com/profile")     # => true
+    #   safe_to_redirect_to?("http://evil.com/profile")        # => false
+    #
+    # Note that the host includes subdomains. A location on the same domain but
+    # a different subdomain to the request will not be considered safe.
+    def safe_to_redirect_to?(location, allow_other_host: _allow_other_host)
+      allow_other_host || _url_host_allowed?(location)
+    end
```

Comment:
Ok, I think I got the alternative version here. I've called it `url_from` to mirror `url_for` except it's going in the opposite direction. The docs may be a little too convoluted, so I'll sleep on those and see if I can whittle them down soon.

```suggestion
    # Extract an internal URL that's safe to redirect to, to allow falling back to an alternative URL in case it's an external potentially-unsafe URL.
    #
    #   redirect_to url_from(params[:redirect_url]) || root_url
    #
    # A URL is considered safe if it's internal, e.g. on the same host as the <tt>request.host</tt>:
    #
    #   # If request.host is example.com:
    #   url_from("https://example.com/profile") # => "https://example.com/profile"
    #   url_from("http://example.com/profile")  # => "http://example.com/profile"
    #   url_from("http://evil.com/profile")     # => nil
    #
    # Subdomains are considered part of the host:
    #
    #   # If request.host is on https://example.com or https://app.example.com, you'd get:
    #   url_from("https://dev.example.com/profile") # => nil
    #
    # NOTE: there's a similarity with <tt>url_for</tt> which generates an internal URL from various options from within the app, e.g. `url_for(@post)`,
    # while <tt>url_from</tt> takes an external parameter to validate the internal URL, e.g. `url_from(params[:redirect_url])`.
    def url_from(location)
      location if _url_host_allowed?(location)
    end
```

Reply:
Haha! We should have warned you before we recorded it, but it was a fantastic PR to discuss! 🙏

Reply:
@dmcge glad you liked it ❤️😄